internal/cloudproviders/aws/main.go

```diff
@@ -264,6 +260,14 @@ func (obj *AwsProvider) Role(resRole consts.KsctlRole) resources.CloudFactory {
 	}
 }
 
+func (obj *AwsProvider) InitScritps(index int) resources.CloudFactory {
+	role := <-obj.chRole
+	randomString, _ := helpers.GenRandomString(5)
+	obj.initScript <- string(role) + strconv.Itoa(index) + randomString
+
+	return obj
+}
+
```

Comment:
I dont understand why we require this function and additional new field for the awsProvider struct

we should before calling the VM create to also pass the scritp as string to the ec2 client as when the vm is created!